systemvm/debian/opt/cloud/bin/setup/init.sh

```diff
@@ -188,6 +188,7 @@ setup_interface_sshd() {
     setup_sshd $ETH1_IP "eth1"
 
   elif [ "$TYPE" == "vpcrouter" ]; then
+    init_interfaces "eth0"
```

Comment:
why not move this entirely to the appliance/type specific script cc @weizhouapache @Pearl1594 ? This would avoid requiring new systemvmtemplate anytime the init (or early-config?) script.

Reply:
With the new approach the IP on the interfaces need to be available right at the time of VM boot up, so that we can perform the scp operation , so, I am not sure moving this to a separate script would eliminated the need to build a new systemVM template.

Reply:
I suppose the question is - is there a way to refactor the script/sequence in a way that avoids building a new template everytime the first-run scripts (such as init.sh or cloud-early-config) are changed? This would improve the maintainability of the systemvmtemplates. @Pearl1594 @weizhouapache

Reply:
@rohityadavcloud 
in my opinion, it might be possible. 
`init.sh` needs to  (1) get cmdline; (2) configure link-local interface; (3) setup ssh. 
The rest (init and configure public and guest nics, etc) can be moved to other scripts and executed after `scp`. 
is it right ? @Pearl1594

Reply:
Yup, if the control IP is set and ssh comes up fine, then yes I guess it could work.